plugin/web/extensions/storytelling/core/hooks.ts

```diff
@@ -0,0 +1,359 @@
+  );
+
+  const saveScene = useCallback((sceneInfo: Omit<Scene, "camera">) => {
+    setScenes(scenes => {
+      const scene = scenes.find(scene => scene.id === sceneInfo.id);
+      if (scene) {
+        scene.title = sceneInfo.title;
+        scene.description = sceneInfo.description;
+      }
+      return [...scenes];
+    });
+  }, []);
+
+  const moveScene = useCallback((dragIndex: number, hoverIndex: number) => {
+    setScenes((prevScenes: Scene[]) =>
+      update(prevScenes, {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, prevScenes[dragIndex] as Scene],
+        ],
+      }),
+    );
+  }, []);
+
+  const clearStory = useCallback(() => {
+    storyId.current = undefined;
+    setScenes([]);
+  }, []);
+
+  const handleEditStory = useCallback(
+    ({ id, scenes }: EditStory["payload"]) => {
+      storyId.current = id;
+      setScenes(scenes ? JSON.parse(scenes) : []);
+      handleSetMode("editor");
+      if (minimized) {
+        handleMinimize();
+      }
+    },
+    [handleSetMode, minimized, handleMinimize],
+  );
+
+  const handleSaveStory = useCallback(
+    ({ id }: SaveStory["payload"]) => {
+      postMsg("saveStoryData", {
+        id,
+        scenes: JSON.stringify(scenes),
+      });
+    },
+    [scenes],
+  );
+
+  const handleDeleteStory = useCallback(({ id }: DeleteStory["payload"]) => {
+    if (storyId.current === id) {
+      storyId.current = undefined;
+      setScenes([]);
+    }
+  }, []);
+
+  const handlePlayStory = useCallback(
+    ({ id, scenes }: PlayStory["payload"]) => {
+      storyId.current = id;
+      setScenes(JSON.parse(scenes ?? "[]"));
+      handleSetMode("player");
+      if (minimized) {
+        handleMinimize();
+      }
+    },
+    [handleSetMode, minimized, handleMinimize],
+  );
+
+  const handleCancelPlayStory = useCallback(
+    ({ id }: CancelPlayStory["payload"]) => {
+      if (storyId.current === id) {
+        storyId.current = undefined;
+        setScenes([]);
+        if (!minimized) {
+          handleMinimize();
+        }
+      }
+    },
+    [minimized, handleMinimize],
+  );
+
+  const shareStory = useCallback(() => {
+    postMsg("shareStory", {
+      scenes: JSON.stringify(scenes),
+    });
+  }, [scenes]);
+
+  useEffect(() => {
+    // mock scenes, only on web dev
+    const mockScenes = false;
+    if (mockScenes) {
+      const scenes = [];
+      for (let i = 1; i <= 3; i += 1) {
+        scenes.push({
+          id: generateId(),
+          title: `Title ${i}`,
+          description: `# Header 1
+## Header 2
+### Header 3
+### Header 4
+### Header 5`,
+          camera: undefined,
+        });
+      }
+      setScenes(scenes);
+    }
+
+    // theme
+    const themeColor = "#00BEBE";
+    document.documentElement.style.setProperty("--theme-color", themeColor);
+    ConfigProvider.config({
+      theme: {
+        primaryColor: themeColor,
+      },
+    });
+
+    // viewport
+    postMsg("getViewport");
+  }, []);
```

Comment:
A couple things/questions:

We have a kind of "global" theme/ directory that is not being overly used yet. But, instead of using AntD's ConfigProvider to provide a shared theme and going this approach, can you use or expand on the theme/ directory. Something like theme color we could set there and then use across extensions (though it is kind of late in the game to do that :P unless of course at the end we have time for refactors).

But also, you set the configprovider here, but I can't see you using primaryColor anywhere. Are you?  It seems like you are using the --theme-color only?

Reply:
--theme-color is used for my styled components to avoid use `#00BEBE` every where.
Antd configProvider setting the theme color is used for some antd components like pagination or <a>, without configProvider it may requires more of hard coded css overwrite, but i think it's also okey to do that.

Reply:
Removed the antd ConfigProvider and use css var overwrite to set the color.

Reply:
I didn't change anything of the theme dirctory since we have already developed some widgets and seems there's no need to make some part of them using common styles, right?

Reply:
About the theme directory, I was thinking more like if you ADDED anything to it, like a common theme color or something, that that would be fine/good and then if there is time for a refactor could use that in other components too.
But at this point its okay.

Reply:
But, I see you replaced setting the config provider theme in the useEffect with `setProperty` calls. Is there a reason you are doing this rather than just doing the overrides inside the styled component's css?
If we have some styles in css files, some styles using the styled component's css string, some styles being set like you are doing here, etc, it gets really hard to troubleshoot style problems.
Styled components allow us to keep most styles in one place, which is really nice and clean. Only occasionally does that not work and we need to do inline styles, and on those occasions that is fine.
If you have any questions feel free to reach out to me on slack/zoom

Reply:
Updated.
I removed all the `setProperty` for the antd variables and move the override into the styled components.